packages/core/src/StateNode.ts

```diff
@@ -271,6 +271,7 @@ export class StateNode<
         ) {
           this.machine.options.actors = {
             ...this.machine.options.actors,
+            // TODO: this should accept `src` as-is
```

Comment:
It would make sense to remove this completely. We shouldn't have to put those on `this.machine.options.actors`. They were not provided there so why we'd keep them there?

This might impact the rehydration story for inline actors but to fix this we can use some extra information to understand how to look those up. We can either encode it in the auto-generated `id` somehow. Or we might have to use some extra `.type`-like thing somewhere.

I would also consider completely disallowing `getPersistedState` when the state has inline actors. How do we rehydrate this?
```ts
createMachine({
  entry: assign((_, __, { spawn }) => ({
    childRef: spawn(fromPromise(() => Promise.resolve(42)))
  }))
})
```

The problem here is that when serializing the persisted state we don't have any way to "point" to a stable behavior implementation of this actor. When rehydrating we won't have a clue how to access the original behavior of that child

Reply:
We also have the same problem with invoke, despite having a "stable" reference to it (object path). And more than that - even referenced implementations have this problem when they are declared using factories.

```ts
createMachine({
  invoke: {
    src: 'dynamic'
  }
}, {
  actors: {
    dynamic: (ctx) => fromPromise(() => Promise.resolve(42))
  }
)
```

How do we rehydrate this? It looks a little bit like maybe `src` should just **never** be a factory. We need to have the ability to use the same actor behavior with different arguments and stuff... but that seems to be solved by `input`! The only thing is that the JS API might look a little bit quirky for this, because we'd have to accept two things for any given actor: 1. a static `src` 2. an input resolver~ (with the "standard" `(ctx, ev) => input` signature)

Reply:
This is removed in `v5/input-3` #3795